prime-router/src/main/kotlin/Report.kt

```diff
@@ -338,6 +359,19 @@ class Report : Logging {
         val combinedSelection = Selection.withRange(0, table.rowCount())
         filterFunctions.forEach { (filterFn, fnArgs) ->
             val filterFnSelection = filterFn.getSelection(fnArgs, table, receiver, doDetailedFilterLogging)
+            if (doDetailedFilterLogging && filterFnSelection.size() < table.rowCount()) {
+                val before = Selection.withRange(0, table.rowCount())
+                filteredRows.add(
+                    FilterResult(
+                        receiver.fullName,
+                        filterFn.name,
+                        fnArgs,
+                        table.rowCount(),
+                        filterFnSelection.size(),
+                        before.andNot(filterFnSelection).toArray()
```

Comment:
If I'm reading this correctly, this will just show me which row numbers were filtered out, but no additional information about them. Does this always correctly map to the index that we can refer back to in the `ItemLineage` table. If I wanted to go back and figure out which rows were washed out, would it be easy to figure that out with this? Should we include the `messageId` field?

Thinking from our side of the work short-term, senders will want to know why records didn't go to a state, and I want to be able to find the details. Long-term I'd like to be able to put this into a dashboard so it's simple for support staff to quickly see and examine what did and did not go where.

If this ties cleanly and consistently to the `parentIndex` field in `ItemLineage`, then we should be good, but I almost want a table that just shows us all the filters that fired for a report.

Reply:
It does match with the 0 based index of the `row` being filtered. So it'll say it filtered rows `1,2,5` etc. Moving to a struct rather than a string leaves room for the more machine readable response in follow up work.

I think adding the `messageId` should be doable from what I'm seeing. Is that both guaranteed to be filled out and enough of a value add to do now?

Reply:
No, there's no guarantee that we get a messageId.   So it would be an 'include it if you have it' feature.

Reply:
The 'consolidated warnings' is also printing out "row" numbers, eg

`  "consolidatedWarnings" : [ {
    "message" : "Invalid code: '3.981E+12' is not a display value in altValues set for 'healthcareEmployeeType' ('healthcare_employee_type').",
    "rows" : "Rows: 7"
  }

We need to make sure you and they are using the same row numbers!

Reply:
There's a [bug here](https://github.com/CDCgov/prime-reportstream/blob/master/prime-router/src/main/kotlin/azure/ReportFunction.kt#L607-L608) where it assumes the type is CSV. How about normalizing it so that we report 1 offset rows. And document that the csv header is not included.